mlir/include/Quantum/IR/QuantumOps.td

```diff
@@ -781,12 +781,15 @@ def AdjointOp : Op<QuantumDialect, "adjoint", [SingleBlock]> {
     let hasVerifier = 1;
 }
 
-def YieldOp : Op<QuantumDialect, "yield", [Pure, ReturnLike, Terminator]> {
+def YieldOp : Op<QuantumDialect, "yield",
+                [Pure, ReturnLike, Terminator, ParentOneOf<["AdjointOp"]>]> {
```

Comment:
The presence of the `pure` tag is interesting. Normally a pure operation (i.e. one with no side-effects) which has no (used) results would be striped from the IR 🤔
But maybe this is not the case with terminators, as the `scf.yield` also appears to be declared pure.

Reply:
I found this [definition](https://github.com/llvm/llvm-project/blob/73a0ae021ec6112911b98025055b8a2e881b1376/mlir/include/mlir/Interfaces/SideEffectInterfaces.td#L133) and the only usage I see in cpp is [here](https://github.com/llvm/llvm-project/blob/73a0ae021ec6112911b98025055b8a2e881b1376/mlir/lib/Dialect/LLVMIR/IR/LLVMInlining.cpp#L321) which is more about inlining than stripping.